### PR TITLE
Add autocomplete suggestions

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will add slash command autocomplete suggestions.

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -181,3 +176,38 @@ func responsef(format string, args ...interface{}) *model.CommandResponse {
...
+	list.AddCommand(optionalListNextWeek)
+	agenda.AddCommand(list)
+
+	queue := model.NewAutocompleteData("queue", "", "Queue `message` as a topic on the next meeting.")
+	queue.AddStaticListArgument("If `next-week` is provided, it will queue for the meeting in the next calendar week.", false, []model.AutocompleteListItem{{
+		HelpText: "If `next-week` is provided, it will queue for the meeting in the next calendar week.",
+		Hint:     "(optional)",
+		Item:     "next-week",
+	}})
+	queue.AddTextArgument("Creates a post for user with the given message for the next meeting date.", "message", "")
```
Comment (jfrerich) [inline server/command.go:194]:
```suggestion
	queue.AddTextArgument("Message for the next meeting date.", "message", "")
```

Comment (jfrerich) [reply server/command.go:194]: The text here is describing the meaning of the `Message` autocomplete option.  I think it makes sense to somehow tell the user `Message` is not an autocomplete value, but an input needed from the user.

On the same point, I wonder if we could/should somehow designate the difference between an autocomplete value versus an input required from the user.  As an example, `next-week` is an autocomplete value and `Message` requires free text from the user.
image

Comment (iomodo) [reply server/command.go:194]: Agree. What we could do is differentiate using icons on the left side, or have different styles, but it is not yet part of the Server PR. I'll create the ticket for this.
I'd suggest using `[message]` for now.

**Comment by iomodo** [comment]
@hanzei There will be no changes for older servers, new functionality will not work of course. No `min_server_version` bump is needed. We need a dependency bump to 5.24 though, to be able to compile the new autocomplete data.

File: server/command.go (hunk 1 above)
Comment (jfrerich) [reply server/command.go]: That'd be perfect.  Thanks!

File: go.mod [hunk 2]
```diff
@@ -4,8 +4,8 @@ go 1.12
 
 require (
 	github.com/blang/semver v3.6.1+incompatible // indirect
-	github.com/mattermost/mattermost-server v1.4.1-0.20191016162522-6597fdb40134 // Mattermost Server 5.16.0
+	github.com/mattermost/mattermost-server/v5 v5.3.2-0.20200525190709-91f010b8b7e8
```
Comment (levb) [inline go.mod:7]: why this version?

Comment (iomodo) [reply go.mod:7]: This was the latest master, we will switch it to 5.24

Comment (hanzei) [reply go.mod:7]: `v5.24-rc1` is out: https://github.com/mattermost/mattermost-server/releases/tag/v5.24.0-rc1

**Comment by DHaussermann** [comment]
I have tested all the auto-complete functionality (except schedule)
- All commands work as expected 
- All argument name and descriptions look correct except schedule (see below)
- Tested the updated plugins on server 5.23.1 to ensure it works as expected without auto complete

@iomodo one question here...  The schedule command functionality has has changed instead of `Mon` or `Monday`  you ave to use `1` Was this part of this PR or do you know anything about it? I can't see a separate PR for it. In my opinion this is less usable. Also in it's current state it's wrong as the description says it will accept `1-5`  however, you can schedule meetings on Saturday and Sunday  with `6` and `7`

**Pull request merged**